nginx/nginx_test.go

```diff
@@ -367,7 +424,7 @@ func TestNginxConfig(t *testing.T) {
 
 		for _, expected := range test.expectedSettings {
 			if strings.HasPrefix(expected, "!") {
-				assert.NotContains(configContents, expected, "%s\nshould not contain setting", test.name)
+				assert.NotContains(configContents, expected[1:], "%s\nshould not contain setting", test.name)
```

Comment:
I don't think this assertion ever worked because, the expected string contained `!` which can never be in the nginx conf and hence was passing incorrectly. The expected string should be sliced to remove the `!` character.